Control/Monad/Logic.hs

```diff
@@ -136,17 +162,33 @@ logic f = LogicT $ \k -> Identity .
                          runIdentity
 
 -------------------------------------------------------------------------
--- | Extracts the first result from a 'Logic' computation.
+-- | Extracts the first result from a 'Logic' computation, failing if
+-- there are no results.
+--
+-- >>> observe (return 5 `mplus` return 3 `mplus` mzero)
+-- 5
+--
+-- >>> observe mzero
+-- *** Exception: No answer.
+--
 observe :: Logic a -> a
 observe lt = runIdentity $ unLogicT lt (const . return) (error "No answer.")
 
 -------------------------------------------------------------------------
 -- | Extracts all results from a 'Logic' computation.
+--
+-- >>> observe (return 5 `mplus` mzero `mplus` mzero `mplus` return 3 `mplus` mzero)
+-- [5,3]
+--
 observeAll :: Logic a -> [a]
 observeAll = runIdentity . observeAllT
 
 -------------------------------------------------------------------------
 -- | Extracts up to a given number of results from a 'Logic' computation.
+--
+-- >>> observeMany 5 nats
```

Comment:
Where is `nats` definition?

Reply:
I was borrowing from a definition above, but I agree it would be better to reproduce it here as well.

Reply:
Let's stick to `Monad` / `MonadPlus` vocabulary, just for the sake of consistency. So ``return 0 `mplus` liftM (1 +) nats``.

(I'd like to switch to `Applicative` / `Alternative` vocabulary in future, but not yet)

Reply:
Done.